tests/helpers/providerStateHelper.js

```diff
@@ -93,19 +161,71 @@ const givenFileFolderIsCreated = (provider, username, password, resource, resour
  *
  * @param {object} provider
  * @param {string} username
- * @param {sring} password
+ * @param {sring} userPassword
  * @param {sring} resource
- * @param {sring} shareType
+ * @param {number} shareType
+ * @param {sring} shareWith
+ * @param {number} permissions
+ * @param {string} expireDate
+ * @param {object} attributes
+ * @param {string} password
+ * @param {boolean} publicUpload
  */
-const givenResourceIsShared = async (provider, username, password, resource, shareType) => {
-  if (shareType === 3) {
-    return givenShareExists(provider, username, password, resource, shareType)
-  } else if (shareType === 0) {
-    await givenUserExists(provider, config.testUser2, config.testUser2Password)
-    return givenShareExists(provider, username, password, resource, shareType, config.testUser2)
-  } else if (shareType === 1) {
-    await givenGroupExists(provider, config.testGroup)
-    return givenShareExists(provider, username, password, resource, shareType, config.testGroup)
+const givenResourceIsShared = async (
+  provider,
+  username,
+  userPassword,
+  resource,
+  shareType,
+  shareWith,
+  permissions,
+  expireDate,
+  attributes,
+  password,
+  publicUpload
+) => {
+  if (shareType === SHARE_TYPE.public) {
```

Comment:
if the shareType given does not match any of your expectations the function will just exit without any error and the developer would have a hard time to debug. Please add some error handling. e.g. throw an exception when an unknown share type is given